src/browser/mod.rs

```diff
@@ -97,22 +97,49 @@ impl Browser {
         )
     }
 
-    //    pub fn new_tab(&self) -> Result<Arc<Tab>, Error> {
-    //        let create_target = target::methods::CreateTarget {
-    //            url: "about:blank",
-    //            width: None,
-    //            height: None,
-    //            browser_context_id: None,
-    //            enable_begin_frame_control: None,
-    //        };
-    //
-    ////        let target_id = self.call_method(create_target)?.target_id;
-    ////        let new_tab = Arc::new(Tab::new(target_id, Arc::clone(&self.transport))?);
-    ////
-    ////        self.add_tab(Arc::clone(&new_tab));
-    //
-    //        Ok(new_tab)
-    //    }
+    /// Create a new tab and return a handle to it.
+    ///
+    /// ```rust
+    /// # use failure::Error;
+    /// # fn main() -> Result<(), Error> {
+    /// #
+    /// # use headless_chrome::{Browser, LaunchOptions};
+    /// # let browser = Browser::new(LaunchOptions::default().unwrap())?;
+    /// let first_tab = browser.wait_for_initial_tab()?;
```

Comment:
What's the significance of waiting for the first new tab here? Do we need one to get a second?

If `wait_for_inititial_tab` does two things, actually waiting for chrome to set itself up and returning the initial tab, I'd suggest we split the API.

Reply:
Good point. It's probably better to have `Browser::new` wait for the first tab to be “discovered” (see https://github.com/atroche/rust-headless-chrome/blob/fb80bb14fe44d261a4efdef00ada2ba8dad23fb9/src/browser/mod.rs#L76) before returning. The delay would likely be unnoticeable, and right now there's a brief window in time where you can have a handle to Browser and have `get_tabs()` return a vector of length zero, which is a bit inconvenient.